actionpack/CHANGELOG.md

```diff
@@ -1,3 +1,41 @@
+*   Raise exception when calling `to_h` in an unpermitted Parameters.
+
+    This method will raise on unpermitted Parameters if
+    `config.action_controller.raise_on_unfiltered_parameters` is true.
+
+    Before we returned either an empty hash or only the always permitted parameters
+    (`:controller` and `:action` by default).
+
+    The previous behavior was dangerous because in order to get the attributes users
+    usually fallback to use `to_unsafe_h that` could potentially introduce security issues.
+
+    *Rafael Mendonça França*
+
+*   Add `ActionController::Parameters#to_hash` to implicit conversion.
+
+    Now methods that implicit convert objects to a hash will be able to work without
+    requiring the users to change their implementation.
+
+    This method will return a `Hash` instead of a `HashWithIndefirentAccess` to mimic the
```

Comment:
I'm curious if we call `symbolize_keys` in `to_hash` would strong parameters basically work as keyword arguments? E.g. being able to do this would be sweet:

```ruby
def hello(a:, d:)
end

hello(ActionController::Parameters.new(a: :c, d: :c))
```

(And I think it would also fix the warts in the new `resolve` and `direct` API around parameters.)